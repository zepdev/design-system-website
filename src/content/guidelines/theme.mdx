---
label: Guidelines
title: Theme
---

# Theme

Themes are used to customize existing elements to fit the specific style of a strategic business unit or product. There is no base color for the Zeppelin-element-library, almost all elements require that a theme class is present. At the moment the Zeppelin Design System offers three main themes. Developers can customize all elements by adding the theme class to any html tag (`<main>`,`<body>`,`<div>`, ...) as long it wraps all elements you want to use.

```js
<main class="theme-zeppelin" id="app-root">
  ...
</main>
```

We currently offer three main themes;

```js
// Zeppelin Indigo Blue
<div class="theme-zeppelin" >...</div>

// CAT Yellow
<div class="theme-cat" >...</div>

// Rental Red
<div class="theme-rental" >...</div>
```
