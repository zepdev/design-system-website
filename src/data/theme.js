module.exports = {
  breakpoints: {
    xs: '36em' /* 576px */,
    s: '48em' /* 768px */,
    m: '62em' /* 992px */,
    l: '75em' /* 1200px */,
    xl: '105em' /* 1680px */,
    xxl: '120em' /* 1920px */,
  },
  typography: {
    typefaces: {
      roboto: "'Roboto', Arial, Helvetica, sans-serif",
      teko: "Teko', sans-serif",
    },
    basics: {
      fontSizeBase: {
        specification: 'font-size-base',
        px: '16px',
        rem: '1rem',
        weight: null,
      },
      lineHeightBase: {
        specification: 'line-height-base',
        px: '24px',
        rem: '1.5rem',
        weight: null,
      },
      fontWeightBase: {
        specification: 'font-weight-base',
        px: null,
        rem: null,
        weight: 400,
      },
    },
    headings: {
      normal: {
        'zep-typo--normal-1': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-2': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-3': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-4': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-5': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-6': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-7': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-8': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
      },
      display: {
        'zep-typo--display-1': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--display-2': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--display-3': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--display-4': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
      },
      special: {
        'zep-typo--special-1': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--special-2': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--special-3': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--special-4': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
      },
    },
  },
  colors: {
    primary: {
      indigoBlue: {
        name: 'indigo-blue',
        hex: '#27166f',
        rgb: '39/22/111',
        hsb: null,
      },
      catYellow: {
        name: 'cat-yellow',
        hex: '#f0b823',
        rgb: '240/184/35',
        hsb: null,
      },
      rentalRed: {
        name: 'rental-red',
        hex: '#af0e0e',
        rgb: '175/14/14',
        hsb: null,
      },
    },
    gray: {
      grayDark: {
        name: 'gray-dark',
        hex: '#101111',
        rgb: '16/17/17',
        hsb: null,
      },
      grayMidDark: {
        name: 'gray-mid-dark',
        hex: '#373d3f',
        rgb: '55/61/63',
        hsb: null,
      },
      gray: { name: 'gray', hex: '#464a4c', rgb: '70/74/76', hsb: null },
      grayLight: {
        name: 'gray-light',
        hex: '#6f7c80',
        rgb: '111/124/128',
        hsb: null,
      },
      grayLighter: {
        name: 'gray-lighter',
        hex: '#eceeef',
        rgb: '236/238/239',
        hsb: null,
      },
      grayMid: {
        name: 'gray-mid',
        hex: '#f2f3f4',
        rgb: '242/243/244',
        hsb: null,
      },
      grayLightest: {
        name: 'gray-lightest',
        hex: '#f7f7f9',
        rgb: '247/247/249',
        hsb: null,
      },
      white: { name: 'white', hex: '#ffffff', rgb: '255/255/255', hsb: null },
    },
    support: {
      alert: { name: 'alert', hex: '#ff6562', rgb: '255/101/98', hsb: null },
      warning: {
        name: 'warning',
        hex: '#f0ad4e',
        rgb: '240,/73/78',
        hsb: null,
      },
      success: { name: 'success', hex: '#5cb85c', rgb: '92/184/92', hsb: null },
      info: { name: 'info', hex: '#5bc0de', rgb: '91/192/222', hsb: null },
    },
  },
  spacing: {
    xxs: 4,
    xs: 8,
    s: 12,
    m: 16,
    l: 24,
    xl: 32,
    xxl: 48,
  },
}
