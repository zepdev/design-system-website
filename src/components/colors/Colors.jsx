import React from 'react'
import PropTypes from 'prop-types'
import withStyles from 'react-jss'
import Swatch from './Swatch'
import { colors } from 'zeppelin-element-library/bundle/themes/theme.json'

const styles = theme => ({
  root: {
    display: 'flex',
    flexWrap: 'wrap',
    justifyContent: 'space-between',
    marginBottom: theme.spacing.xl.px,
  },
})

function Colors({ section, classes }) {
  return (
    <div className={classes.root}>
      {Object.keys(colors[section]).map((elem, idx) => (
        <Swatch color={colors[section][elem]} section={section} key={`swatch${ idx }`} />
      ))}
    </div>
  )
}

Swatch.propTypes = {
  classes: PropTypes.object,
  section: PropTypes.string.isRequired,
}

export default withStyles(styles)(Colors)
