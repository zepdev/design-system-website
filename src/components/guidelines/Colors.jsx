import React, { useState, Fragment } from 'react'
import PropTypes from 'prop-types'
import withStyles from 'react-jss'
import Swatch from '../swatch/Swatch'
import Tabs from '../tabs/Tabs'
import Tab from '../tabs/Tab'
import Headline from '../typography/Headline'
import ElementUsage from '../elements/ElementUsage'
import { colors } from 'zeppelin-element-library/bundle/themes/theme.json'

const styles = theme => ({
  tabs: {
    marginBottom: `${theme.spacing.component.xxl.rem}rem`,
  },
  container: {
    display: 'flex',
    flexWrap: 'wrap',
    justifyContent: 'space-between',
  },
  [`@media (min-width: ${theme.breakpoints.m})`]: {
    container: {
      justifyContent: 'flex-start',
    },
  },
})

function Colors({ classes }) {
  const [value, setValue] = useState(0)

  function handleChange(newValue) {
    setValue(newValue)
  }

  return (
    <>
      <Tabs value={value} onClick={handleChange} className={classes.tabs}>
        <Tab label="Overview" />
        <Tab label="Usage" />
      </Tabs>
      {value === 0 && (
        <>
          {Object.keys(colors).map((section, idx) => (
            <Fragment key={`swatch${idx}`}>
              <Headline variant="md">{section}</Headline>
              <div className={classes.container}>
                {Object.keys(colors[section]).map(elem => (
                  <Swatch
                    color={colors[section][elem]}
                    section={section}
                    key={`swatch${elem}`}
                  />
                ))}
              </div>
            </Fragment>
          ))}
        </>
      )}
      {value === 1 && <ElementUsage element="colors" />}
    </>
  )
}

Swatch.propTypes = {
  classes: PropTypes.object,
}

export default withStyles(styles)(Colors)
