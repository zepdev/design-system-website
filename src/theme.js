import { createMuiTheme } from '@material-ui/core/styles'
import guidelines from '@zlab-de/zel-react/guidelines.json'

const theme = createMuiTheme({
  palette: {
    type: 'light',
    primary: {
      light: guidelines.theme.indigo.secondary,
      main: guidelines.theme.indigo.primary,
      dark: guidelines.theme.indigo.primary,
    },
    error: { main: guidelines.color.support.alert },
  },
  status: {
    gray: '#eceeef',
  },
  logo: guidelines.logo,
  blue: guidelines.theme.blue,
  indigo: guidelines.theme.indigo,
  red: guidelines.theme.red,
  yellow: guidelines.theme.yellow,
  guidelines: guidelines,
  space: guidelines.spacing.component,
  color: guidelines.color,
})

export default theme
