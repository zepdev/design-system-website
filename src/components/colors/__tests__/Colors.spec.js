import React from 'react'
import { render } from 'react-testing-library'
import { ThemeProvider } from 'react-jss'
import theme from '../../../data/theme'
import Colors from '../Colors'

describe('Colors', () => {
  it('renders correctly', () => {
    const mockProps = {
      classes: {},
      section: 'primary',
    }
    const component = render(
      <ThemeProvider theme={theme}>
        <Colors {...mockProps} />
      </ThemeProvider>
    )
    expect(component).toMatchSnapshot()
  })
})
