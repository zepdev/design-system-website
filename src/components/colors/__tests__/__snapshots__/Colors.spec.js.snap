// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Colors renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Colors-root-0-1-1"
      >
        <div
          class="Swatch-root-0-1-2 Swatch-third-0-1-3"
        >
          <button
            class="ButtonBase-button-0-1-12 zep-button Swatch-button-0-1-5"
            style="background: rgb(39, 22, 111);"
          >
            <span
              class="Swatch-backdrop-0-1-6"
            />
            <span
              class="Swatch-backdropContent-0-1-7"
            >
              <svg
                aria-labelledby="title"
                class="Swatch-icon-0-1-8"
                height="32"
                viewBox="0 0 24 24"
                width="32"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title
                  id="title_copy"
                >
                  copy hex:#27166f
                </title>
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
                <path
                  d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm-1 4H8c-1.1 0-1.99.9-1.99 2L6 21c0 1.1.89 2 1.99 2H19c1.1 0 2-.9 2-2V11l-6-6zM8 21V7h6v5h5v9H8z"
                  fill="currentColor"
                />
              </svg>
              <span
                class="Swatch-buttonText-0-1-9"
                data-testid="swatch-text"
              >
                #27166f
              </span>
            </span>
          </button>
          <p
            class="Swatch-name-0-1-11 zep-typo--normal-2"
          >
            indigo-blue
          </p>
          <p
            class="Swatch-text-0-1-10 zep-typo--normal-1"
          >
            HEX: #27166f
          </p>
          <p
            class="Swatch-text-0-1-10 zep-typo--normal-1"
          >
            RGB: 39/22/111
          </p>
        </div>
        <div
          class="Swatch-root-0-1-2 Swatch-third-0-1-3"
        >
          <button
            class="ButtonBase-button-0-1-12 zep-button Swatch-button-0-1-5"
            style="background: rgb(240, 184, 35);"
          >
            <span
              class="Swatch-backdrop-0-1-6"
            />
            <span
              class="Swatch-backdropContent-0-1-7"
            >
              <svg
                aria-labelledby="title"
                class="Swatch-icon-0-1-8"
                height="32"
                viewBox="0 0 24 24"
                width="32"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title
                  id="title_copy"
                >
                  copy hex:#f0b823
                </title>
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
                <path
                  d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm-1 4H8c-1.1 0-1.99.9-1.99 2L6 21c0 1.1.89 2 1.99 2H19c1.1 0 2-.9 2-2V11l-6-6zM8 21V7h6v5h5v9H8z"
                  fill="currentColor"
                />
              </svg>
              <span
                class="Swatch-buttonText-0-1-9"
                data-testid="swatch-text"
              >
                #f0b823
              </span>
            </span>
          </button>
          <p
            class="Swatch-name-0-1-11 zep-typo--normal-2"
          >
            cat-yellow
          </p>
          <p
            class="Swatch-text-0-1-10 zep-typo--normal-1"
          >
            HEX: #f0b823
          </p>
          <p
            class="Swatch-text-0-1-10 zep-typo--normal-1"
          >
            RGB: 240/184/35
          </p>
        </div>
        <div
          class="Swatch-root-0-1-2 Swatch-third-0-1-3"
        >
          <button
            class="ButtonBase-button-0-1-12 zep-button Swatch-button-0-1-5"
            style="background: rgb(175, 14, 14);"
          >
            <span
              class="Swatch-backdrop-0-1-6"
            />
            <span
              class="Swatch-backdropContent-0-1-7"
            >
              <svg
                aria-labelledby="title"
                class="Swatch-icon-0-1-8"
                height="32"
                viewBox="0 0 24 24"
                width="32"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title
                  id="title_copy"
                >
                  copy hex:#af0e0e
                </title>
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
                <path
                  d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm-1 4H8c-1.1 0-1.99.9-1.99 2L6 21c0 1.1.89 2 1.99 2H19c1.1 0 2-.9 2-2V11l-6-6zM8 21V7h6v5h5v9H8z"
                  fill="currentColor"
                />
              </svg>
              <span
                class="Swatch-buttonText-0-1-9"
                data-testid="swatch-text"
              >
                #af0e0e
              </span>
            </span>
          </button>
          <p
            class="Swatch-name-0-1-11 zep-typo--normal-2"
          >
            rental-red
          </p>
          <p
            class="Swatch-text-0-1-10 zep-typo--normal-1"
          >
            HEX: #af0e0e
          </p>
          <p
            class="Swatch-text-0-1-10 zep-typo--normal-1"
          >
            RGB: 175/14/14
          </p>
        </div>
        <div
          class="Swatch-root-0-1-2 Swatch-third-0-1-3"
        >
          <button
            class="ButtonBase-button-0-1-12 zep-button Swatch-button-0-1-5"
            style="background: rgb(0, 174, 239);"
          >
            <span
              class="Swatch-backdrop-0-1-6"
            />
            <span
              class="Swatch-backdropContent-0-1-7"
            >
              <svg
                aria-labelledby="title"
                class="Swatch-icon-0-1-8"
                height="32"
                viewBox="0 0 24 24"
                width="32"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title
                  id="title_copy"
                >
                  copy hex:#00aeef
                </title>
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
                <path
                  d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm-1 4H8c-1.1 0-1.99.9-1.99 2L6 21c0 1.1.89 2 1.99 2H19c1.1 0 2-.9 2-2V11l-6-6zM8 21V7h6v5h5v9H8z"
                  fill="currentColor"
                />
              </svg>
              <span
                class="Swatch-buttonText-0-1-9"
                data-testid="swatch-text"
              >
                #00aeef
              </span>
            </span>
          </button>
          <p
            class="Swatch-name-0-1-11 zep-typo--normal-2"
          >
            zlab-blue
          </p>
          <p
            class="Swatch-text-0-1-10 zep-typo--normal-1"
          >
            HEX: #00aeef
          </p>
          <p
            class="Swatch-text-0-1-10 zep-typo--normal-1"
          >
            RGB: 0/174/239
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Colors-root-0-1-1"
    >
      <div
        class="Swatch-root-0-1-2 Swatch-third-0-1-3"
      >
        <button
          class="ButtonBase-button-0-1-12 zep-button Swatch-button-0-1-5"
          style="background: rgb(39, 22, 111);"
        >
          <span
            class="Swatch-backdrop-0-1-6"
          />
          <span
            class="Swatch-backdropContent-0-1-7"
          >
            <svg
              aria-labelledby="title"
              class="Swatch-icon-0-1-8"
              height="32"
              viewBox="0 0 24 24"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title
                id="title_copy"
              >
                copy hex:#27166f
              </title>
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm-1 4H8c-1.1 0-1.99.9-1.99 2L6 21c0 1.1.89 2 1.99 2H19c1.1 0 2-.9 2-2V11l-6-6zM8 21V7h6v5h5v9H8z"
                fill="currentColor"
              />
            </svg>
            <span
              class="Swatch-buttonText-0-1-9"
              data-testid="swatch-text"
            >
              #27166f
            </span>
          </span>
        </button>
        <p
          class="Swatch-name-0-1-11 zep-typo--normal-2"
        >
          indigo-blue
        </p>
        <p
          class="Swatch-text-0-1-10 zep-typo--normal-1"
        >
          HEX: #27166f
        </p>
        <p
          class="Swatch-text-0-1-10 zep-typo--normal-1"
        >
          RGB: 39/22/111
        </p>
      </div>
      <div
        class="Swatch-root-0-1-2 Swatch-third-0-1-3"
      >
        <button
          class="ButtonBase-button-0-1-12 zep-button Swatch-button-0-1-5"
          style="background: rgb(240, 184, 35);"
        >
          <span
            class="Swatch-backdrop-0-1-6"
          />
          <span
            class="Swatch-backdropContent-0-1-7"
          >
            <svg
              aria-labelledby="title"
              class="Swatch-icon-0-1-8"
              height="32"
              viewBox="0 0 24 24"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title
                id="title_copy"
              >
                copy hex:#f0b823
              </title>
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm-1 4H8c-1.1 0-1.99.9-1.99 2L6 21c0 1.1.89 2 1.99 2H19c1.1 0 2-.9 2-2V11l-6-6zM8 21V7h6v5h5v9H8z"
                fill="currentColor"
              />
            </svg>
            <span
              class="Swatch-buttonText-0-1-9"
              data-testid="swatch-text"
            >
              #f0b823
            </span>
          </span>
        </button>
        <p
          class="Swatch-name-0-1-11 zep-typo--normal-2"
        >
          cat-yellow
        </p>
        <p
          class="Swatch-text-0-1-10 zep-typo--normal-1"
        >
          HEX: #f0b823
        </p>
        <p
          class="Swatch-text-0-1-10 zep-typo--normal-1"
        >
          RGB: 240/184/35
        </p>
      </div>
      <div
        class="Swatch-root-0-1-2 Swatch-third-0-1-3"
      >
        <button
          class="ButtonBase-button-0-1-12 zep-button Swatch-button-0-1-5"
          style="background: rgb(175, 14, 14);"
        >
          <span
            class="Swatch-backdrop-0-1-6"
          />
          <span
            class="Swatch-backdropContent-0-1-7"
          >
            <svg
              aria-labelledby="title"
              class="Swatch-icon-0-1-8"
              height="32"
              viewBox="0 0 24 24"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title
                id="title_copy"
              >
                copy hex:#af0e0e
              </title>
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm-1 4H8c-1.1 0-1.99.9-1.99 2L6 21c0 1.1.89 2 1.99 2H19c1.1 0 2-.9 2-2V11l-6-6zM8 21V7h6v5h5v9H8z"
                fill="currentColor"
              />
            </svg>
            <span
              class="Swatch-buttonText-0-1-9"
              data-testid="swatch-text"
            >
              #af0e0e
            </span>
          </span>
        </button>
        <p
          class="Swatch-name-0-1-11 zep-typo--normal-2"
        >
          rental-red
        </p>
        <p
          class="Swatch-text-0-1-10 zep-typo--normal-1"
        >
          HEX: #af0e0e
        </p>
        <p
          class="Swatch-text-0-1-10 zep-typo--normal-1"
        >
          RGB: 175/14/14
        </p>
      </div>
      <div
        class="Swatch-root-0-1-2 Swatch-third-0-1-3"
      >
        <button
          class="ButtonBase-button-0-1-12 zep-button Swatch-button-0-1-5"
          style="background: rgb(0, 174, 239);"
        >
          <span
            class="Swatch-backdrop-0-1-6"
          />
          <span
            class="Swatch-backdropContent-0-1-7"
          >
            <svg
              aria-labelledby="title"
              class="Swatch-icon-0-1-8"
              height="32"
              viewBox="0 0 24 24"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title
                id="title_copy"
              >
                copy hex:#00aeef
              </title>
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm-1 4H8c-1.1 0-1.99.9-1.99 2L6 21c0 1.1.89 2 1.99 2H19c1.1 0 2-.9 2-2V11l-6-6zM8 21V7h6v5h5v9H8z"
                fill="currentColor"
              />
            </svg>
            <span
              class="Swatch-buttonText-0-1-9"
              data-testid="swatch-text"
            >
              #00aeef
            </span>
          </span>
        </button>
        <p
          class="Swatch-name-0-1-11 zep-typo--normal-2"
        >
          zlab-blue
        </p>
        <p
          class="Swatch-text-0-1-10 zep-typo--normal-1"
        >
          HEX: #00aeef
        </p>
        <p
          class="Swatch-text-0-1-10 zep-typo--normal-1"
        >
          RGB: 0/174/239
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
